src/analyze/analyze-unit-gdb-start.c

```diff
@@ -0,0 +1,241 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <stdlib.h>
+#include <sys/socket.h>
+#include <unistd.h>
+
+#include "sd-bus.h"
+
+#include "alloc-util.h"
+#include "analyze.h"
+#include "analyze-unit-gdb-start.h"
+#include "analyze-verify-util.h"
+#include "bus-error.h"
+#include "bus-util.h"
+#include "bus-locator.h"
+#include "bus-wait-for-units.h"
+#include "bus-wait-for-jobs.h"
+#include "log.h"
+#include "manager.h"
+#include "pidref.h"
+#include "process-util.h"
+#include "runtime-scope.h"
+#include "service.h"
+#include "signal-util.h"
+#include "strv.h"
+#include "unit.h"
+#include "unit-name.h"
+
+static int start_and_wait_unit(
+                sd_bus *bus,
+                const char *method,
+                const char *name,
+                const char *mode,
+                sd_bus_error *error,
+                BusWaitForJobs *w) {
+
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *reply = NULL;
+        const char *path;
+        int r;
+
+        assert(method);
+        assert(name);
+        assert(mode);
+        assert(error);
+
+        log_debug("%s dbus call org.freedesktop.systemd1.Manager %s(%s, %s)",
+                  "Executing", method, name, mode);
+
+        r = bus_call_method(bus, bus_systemd_mgr, method, error, &reply, "ss", name, mode);
+        if (r < 0)
+                return log_error_errno(r, "Failed to start %s: %s", name, bus_error_message(error, r));
+
+        r = sd_bus_message_read(reply, "o", &path);
+        if (r < 0)
+                return bus_log_parse_error(r);
+        if (w) {
+                log_debug("Adding %s to the set", path);
+                r = bus_wait_for_jobs_add(w, path);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to watch job for %s: %m", name);
+        }
+
+        r = bus_wait_for_jobs_one(w, path, /* flags= */ BUS_WAIT_JOBS_LOG_ERROR, /* extra_args= */ NULL);
+        if (r < 0)
+                return r;
+
+        return 0;
+}
+
+int verb_unit_gdb_start(int argc, char *argv[], void *userdata) {
+        static const struct sigaction sa = {
+                .sa_sigaction = sigterm_process_group_handler,
+                .sa_flags = SA_SIGINFO,
+        };
+
+        _cleanup_free_ char *unit = NULL;
+        _cleanup_(manager_freep) Manager *m = NULL;
+        Unit *u;
+        int r;
+
+        if (arg_transport != BUS_TRANSPORT_LOCAL)
+                return log_error_errno(SYNTHETIC_ERRNO(EINVAL), "Cannot spawn a debugger for a remote service");
+
+        r = unit_name_mangle_with_suffix(argv[1], "as unit", UNIT_NAME_MANGLE_WARN, ".service", &unit);
+        if (r < 0)
+                return log_error_errno(r, "Cannot mangle unit name: %m");
+
+        r = manager_new(RUNTIME_SCOPE_USER, MANAGER_TEST_RUN_MINIMAL, &m);
```

Comment:
This just creates a local mock, so it doesn't do what is needed. You need to query the properties over D-Bus instead, to check what the actual loaded settings are